core/src/main/java/me/lokka30/treasury/plugin/core/utils/UpdateChecker.java

```diff
@@ -72,65 +129,101 @@ private static void handleVersionCheck(String readString) {
             EconomyAPIVersion latestVersion = EconomyAPIVersion.getCurrentAPIVersion();
 
             handlePluginVersioning(plugin, comparisonResult, latestVersionRead);
-
-            PluginVersion.ComparisonResult apiComparisonResult = Utils.compareAPIVersions(providerVersion,
-                    latestVersion
+            handleAPIVersioning(providerVersion,
+                    latestVersion,
+                    providerEconomyProvider,
+                    comparisonResult,
+                    currentVersion.isDevelopmentVersion()
             );
-            if (apiComparisonResult == PluginVersion.ComparisonResult.OLDER) {
-                // this means that "providerVersion" is older than the latest version
-                plugin
-                        .logger()
-                        .error("Economy provider going by the plugin name '" + providerEconomyProvider
-                                .registrar()
-                                .getName() + "'" + "is utilising an older version of the Treasury Economy API");
-                plugin.logger().error(
-                        "than what your current version of Treasury provides. You should inform the author(s)");
-                plugin.logger().error(
-                        "of that plugin that they should update their resource to use the newer Treasury Economy API");
-                plugin.logger().error(" ");
-                if (comparisonResult == PluginVersion.ComparisonResult.NEWER) {
+        }
+    }
+
+    private static void handleGitHubVersioning(PluginVersion.ComparisonResult comparisonResult) {
+        TreasuryPlugin plugin = TreasuryPlugin.getInstance();
+        if (comparisonResult == PluginVersion.ComparisonResult.NEWER) {
+            // this statement means that the "latestVersionRead" is newer than the version we're running.
+            plugin.logger().warn(
+                    "You are running an outdated development version. There may be issues that have been fixed in the newer development versions. Please update as soon as possible.");
+        } else if (comparisonResult == PluginVersion.ComparisonResult.OLDER) {
+            // this statement means that the "latestVersionRead" is older than the version we're running.
+            plugin
+                    .logger()
+                    .warn("You are running a newer development version. How did we get here?");
+        } else if (comparisonResult == PluginVersion.ComparisonResult.EQUAL) {
+            plugin.logger().warn(
+                    "You are running a development version of Treasury. If there are any issues, please report them to: https://github.com/lokka30/Treasury/issues .");
+            plugin.logger().error("USE WITH CAUTION");
```

Comment:
I can understand the warn message but why is "USE WITH COUTION" at the error level? It makes more sense on the warn level

Reply:
I've seen that you use more times that technique. Since color coding seems to be supported by the logger (in the line 246 of this file color coding is ussed) couldn't you color code the message?

Reply:
Why do you care so much about what log level is used? Isn't it better to have the entire line colored in red rather than a part of it?

Reply:
log levels indicate the severity of the message. They are used to inform not to color code

Reply:
my experiments conclude that server owners/administrators tend to be taken more by the color of the message rather than the log level.